src/Symfony/Bundle/FrameworkBundle/Tests/Console/ApplicationTest.php

```diff
@@ -258,14 +258,31 @@ private function getKernel(array $bundles, $useDispatcher = false)
         $container
             ->expects($this->exactly(2))
             ->method('hasParameter')
-            ->withConsecutive(['console.command.ids'], ['console.lazy_command.ids'])
-            ->willReturnOnConsecutiveCalls(true, true)
+            ->willReturnCallback(function (...$args) {
+                static $series = [
+                    [['console.command.ids'], true],
+                    [['console.lazy_command.ids'], true],
+                ];
+
+                [$expectedArgs, $return] = array_shift($series);
+
+                return $expectedArgs === $args ? $return : $this->fail();
+            })
```

Comment:
this is exactly what `willReturnMap` method does. IMO reusing phpunit's method would be better approach

Reply:
I am not sure that `willReturnMap()` takes into account the order in which arguments are called :thinking:

Reply:
You can test this out by using `willReturnMap` an shuffling the arguments, right?

Reply:
Alright so I tried and `willReturnMap()` doesn't check the call order. Tests using `withConsecutive()` need to check the call order so I guess Nicolas' solution is more relevant in this case :slightly_smiling_face:

Reply:
Good catch! Didnt know that return map doesn't check the call order :)